### ISSUE TITLE
Still not found app-updte.yml with nsis.

### ISSUE THREAD

**Issue by cmzz** [electron-userland/electron-builder#1939]
* **Version**: 19.16.0

* **Target**: WIN32

Build with nsis. Installed app,cannot found app-update.yml

```
[2017-08-12 20:50:45:0268] [info] 0.2.3
[2017-08-12 20:50:45:0312] [info] Checking for update
[2017-08-12 20:50:45:0559] [error] Error: Error: ENOENT: no such file or directory, open 'C:\Program Files (x86)\pigtk-robot\resources\app-update.yml'
    at C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\src\AppUpdater.ts:304:27
    at Generator.next (<anonymous>)
From previous event:
    at NsisUpdater.loadUpdateConfig (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\out\AppUpdater.js:338:11)
    at Lazy.AppUpdater.configOnDisk.Lazy [as creator] (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\src\AppUpdater.ts:81:53)
    at Lazy.get value [as value] (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-builder-http\src\httpExecutor.ts:307:23)
    at C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\src\AppUpdater.ts:211:33
    at Generator.next (<anonymous>)
From previous event:
    at NsisUpdater.doCheckForUpdates (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\out\AppUpdater.js:295:11)
    at C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\src\AppUpdater.ts:201:25
    at Generator.next (<anonymous>)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at NsisUpdater._checkForUpdates (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\out\AppUpdater.js:254:11)
    at NsisUpdater.checkForUpdates (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\src\AppUpdater.ts:164:35)
    at t.a (C:\Program Files (x86)\pigtk-robot\resources\app\dist\electron\main.js:1:24911)
    at App.<anonymous> (C:\Program Files (x86)\pigtk-robot\resources\app\dist\electron\main.js:1:36326)
    at emitTwo (events.js:111:20)
    at App.emit (events.js:194:7)
```

package.json:
```
"build": {
    "asar": false,
    "productName": "pigtk-robot",
    "appId": "org.simulatedgreg.electron-vue",
    "directories": {
      "output": "build"
    },
    "files": [
      "dist/electron/**/*"
    ],
    "dmg": {
      "contents": [
        {
          "x": 410,
          "y": 150,
          "type": "link",
          "path": "/Applications"
        },
        {
          "x": 130,
          "y": 150,
          "type": "file"
        }
      ]
    },
    "mac": {
      "icon": "build/icons/icon.icns"
    },
    "win": {
      "icon": "build/icons/icon.ico",
      "publish": [

      ],
      "target": "nsis",
      "certificateFile": "./certs/pigtk-robot.pfx",
      "certificatePassword": "my_password"
    }
    ],
    "nsis": {
      "oneClick": true,
      "perMachine": true,
      "runAfterFinish": true,
      "createDesktopShortcut": true,
      "warningsAsErrors": false
    }
```

**Comment by develar** [comment]
Please set env `DEBUG=electron-builder` and attach log of the terminal output.

**Comment by cmzz** [comment]
Oh.. The compilation lasted for 30 minutes and continued.

**Comment by develar** [comment]
It seems it is npm debug problem. Ok, terminate.

I think problem is that your 

```
"publish": [

      ],
```

is empty. Please remove or configure properly.

**Comment by cmzz** [comment]
![image](https://user-images.githubusercontent.com/6111715/29241308-93cae464-7fa9-11e7-94bc-576ea23aa766.png)


@develar What happened... The log file has a size of 360MB

**Comment by develar** [comment]
`set DEBUG=electron-builder` not `*` ;)

**Comment by cmzz** [comment]
I'v set publish and try it again.

    ```
"win": {
      "icon": "build/icons/icon.ico",
      "publish": [
        "github"
      ],
```

**Comment by cmzz** [comment]
👌，i will try it again..

**Comment by cmzz** [comment]
[ConEmu-2017-08-12-p3292.txt](https://github.com/electron-userland/electron-builder/files/1219839/ConEmu-2017-08-12-p3292.txt)

@develar  Log file is here

**Comment by cmzz** [comment]
Another error this time.

```
[2017-08-12 22:19:48:0722] [info] Generated new staging user ID: 781f77fc-5ebd-55f6-8680-b863dfc27495
[2017-08-12 22:19:50:0153] [error] Error: Error: Update info doesn't contain sha2 or sha512 checksum: "undefined"
    at Function.validateUpdateInfo (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\src\main.ts:66:13)
    at C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\src\GenericProvider.ts:41:14
From previous event:
    at GenericProvider.getLatestVersion (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\out\GenericProvider.js:79:11)
    at C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\src\AppUpdater.ts:217:38
From previous event:
    at NsisUpdater.doCheckForUpdates (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\out\AppUpdater.js:295:11)
    at C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\src\AppUpdater.ts:201:25
    at Generator.next (<anonymous>)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at NsisUpdater._checkForUpdates (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\out\AppUpdater.js:254:11)
    at NsisUpdater.checkForUpdates (C:\Program Files (x86)\pigtk-robot\resources\app\node_modules\electron-updater\src\AppUpdater.ts:164:35)
    at t.a (C:\Program Files (x86)\pigtk-robot\resources\app\dist\electron\main.js:1:24911)
    at App.<anonymous> (C:\Program Files (x86)\pigtk-robot\resources\app\dist\electron\main.js:1:36326)
    at emitTwo (events.js:111:20)
    at App.emit (events.js:194:7)
```

**Comment by develar** [comment]
Interesting, I see 

```
  electron-builder publish:
  electron-builder   - provider: generic
  electron-builder     url: 'http://release-server.pigtk.com/update/${os}/${version}'
````

Ok, more debug —  set `set DEBUG=electron-builder*`

**Comment by develar** [comment]
Stop — do you mean that now app-update.yml correctly created? And you have another issue?

**Comment by cmzz** [comment]
At the server , i used   electron-release-server

**Comment by cmzz** [comment]
En. The app-update.yml correctly created

**Comment by develar** [comment]
Probably you need to use fork https://github.com/jmvallejo/electron-release-server

Please see https://github.com/ArekSredzki/electron-release-server/pull/128

**Comment by develar** [comment]
@cmzz electron-updater is serverless and you don't need electron-release-server

**Comment by cmzz** [comment]
Don't need server? How do I publish the new release.

**Comment by cmzz** [comment]
Im in China. As you know, can't visit github.com without VPN.. So I cant use github or s3 to publish production.

**Comment by develar** [comment]
You can access your data through CloudFront. The Amazon CloudFront Global Edge Network included China too. If you company, it is better for you to avoid maintaining own server. Also, it seems there is China regions for S3.

**Comment by cmzz** [comment]
Thanks. I've Changed to use s3. 

`[2017-08-13 00:48:16:0785] [info] Update for version 0.2.3 is not available (latest version: 0.2.3, downgrade is disallowed.`

Is that means the autoUpdater is ok?

**Comment by develar** [comment]
> Is that means the autoUpdater is ok?

Yes.
